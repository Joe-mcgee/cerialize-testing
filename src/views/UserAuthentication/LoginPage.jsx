import React from "react";
import PropTypes from "prop-types";
// material-ui components
import withStyles from "material-ui/styles/withStyles";
import InputAdornment from "material-ui/Input/InputAdornment";
// material-ui-icons
import Email from "material-ui-icons/Email";
import LockOutline from "material-ui-icons/LockOutline";
// core components
import GridContainer from "../../components/Grid/GridContainer.jsx";
import ItemGrid from "../../components/Grid/ItemGrid.jsx";
import LoginCard from "../../components/Cards/LoginCard.jsx";
import CustomInput from "../../components/CustomInput/CustomInput.jsx";
import Button from "../../components/CustomButtons/Button.jsx";
import loginPageStyle from "../../assets/jss/material-dashboard-pro-react/views/loginPageStyle.jsx";
import firebaseProviders from "../../routes/firebase-socials/index.js";

class LoginPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      cardAnimaton: ""
    };
  }
  componentDidMount() {
    setTimeout(
      function() {
        this.setState({ cardAnimaton: "" });
      }.bind(this),
      700
    );
  }
  popupLogin(provider) {
    return firebaseProviders.popupLogin();
  }
  render() {
    const { classes } = this.props;
    return (
      <div className={classes.content}>
        <div className={classes.container}>
          <GridContainer justify="center">
            <ItemGrid xs={12} sm={6} md={4}>
              <form>
                <LoginCard
                  headerColor="rose"
                  cardTitle="Account Login"
                  footerAlign="center"
                  footer={
                    <Button color="rose" wd size="md">
                      Login
                    </Button>
                  }
                  socials={[
                    "fab fa-facebook-square",
                    "fab fa-google-plus"
                  ].map((prop, key) => {
                    return (
                      <Button
                        color="simple"
                        justIcon
                        key={key}
                        customClass={classes.customButtonClass}
                      >
                        <i className={prop} />
                      </Button>
                    );
                  })}
                  content={
                    <div>
                      <CustomInput
                        labelText="Email"
                        id="email"
                        formControlProps={{
                          fullWidth: true
                        }}
                        inputProps={{
                          endAdornment: (
                            <InputAdornment position="end">
                              <Email className={classes.inputAdornmentIcon} />
                            </InputAdornment>
                          )
                        }}
                      />
                      <CustomInput
                        labelText="Password"
                        id="password"
                        formControlProps={{
                          fullWidth: true
                        }}
                        inputProps={{
                          endAdornment: (
                            <InputAdornment position="end">
                              <LockOutline
                                className={classes.inputAdornmentIcon}
                              />
                            </InputAdornment>
                          )
                        }}
                      />
                    </div>
                  }
                />
              </form>
            </ItemGrid>
          </GridContainer>
        </div>
      </div>
    );
  }
}

LoginPage.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(loginPageStyle)(LoginPage);
