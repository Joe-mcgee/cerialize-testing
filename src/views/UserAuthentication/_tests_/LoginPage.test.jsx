import React from 'react';
import LoginPage from '../LoginPage';

/*test Dependencies*/
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import { configure } from 'enzyme';
import shallow from 'enzyme';
import { mount } from 'enzyme';
import render from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import expect from 'expect';
import enzymify from 'expect-enzyme'
import sinon from 'sinon';

/*material-ui component dependencies*/
import withStyles from "material-ui/styles/withStyles";
import InputAdornment from "material-ui/Input/InputAdornment";
import Face from "material-ui-icons/Face";
import Email from "material-ui-icons/Email";
import LockOutline from "material-ui-icons/LockOutline";

/*core component dependencies*/
import GridContainer from "../../../components/Grid/GridContainer.jsx";
import ItemGrid from "../../../components/Grid/ItemGrid.jsx";
import LoginCard from "../../../components/Cards/LoginCard.jsx";
import CustomInput from "../../../components/CustomInput/CustomInput.jsx";
import Button from "../../../components/CustomButtons/Button.jsx";
import loginPageStyle from "../../../assets/jss/material-dashboard-pro-react/views/loginPageStyle.jsx";

Enzyme.configure({ adapter: new Adapter() });

describe(' LoginPage Tests ', 
 (/* REPO: https://github.com/jesseroyc/cerialize
*/) => {/* Login Page - Origin Branch - Has Three Sections */

/*BUISNESS LOGIC - CONTROLLERS*/
 
  describe(' Buisness Logic ',
     (/* REPO: https://github.com/jesseroyc/cerialize
    */) => {/* Buisness Logic - Controllers - Has Three Components */

    /*User Login Auth*/
     
      describe(' User Login Authentication Features',
       (/* REPO: https://github.com/jesseroyc/cerialize
      */) => {/* User Login Authorization - Has Four Features */
      
          it(' 1) Verified Users Get Cookies ', 
           (/* REPO: https://github.com/jesseroyc/cerialize
          */) => {/* secure encrypted password verified, cookie issued*/
            
            /*
             *code
             */
            
            expect(true).toEqual(true);
          });
          it(' 2) Verified Users Get Sessions', 
           (/* REPO: https://github.com/jesseroyc/cerialize
          */) => {/* secure encrypted password verified, session issued*/
            
            /*
             *code
             */
            
            expect(true).toEqual(true);
          });
          it(' 3) Text Field Security ', 
           (/* REPO: https://github.com/jesseroyc/cerialize
          */) => {/* user attempts malicious tampering of input fields*/
            
            /*
             *code
             */
            
            expect(true).toEqual(true);
          });
      });

    /*Social Network Login*/
     
      describe(' Social Network Login Successful ',
       (/* REPO: https://github.com/jesseroyc/cerialize
      */) => {/* Social Network Login - Has Three Features*/
      
          it(' 1) Login With Twitter ', 
           (/* REPO: https://github.com/jesseroyc/cerialize
          */) => {/* user logs in using Twitter account credentials*/
            
            /*
             *code
             */
            
            expect(true).toEqual(true);
          });
          it(' 2) Login With Facebook ', 
           (/* REPO: https://github.com/jesseroyc/cerialize
          */) => {/* user logs in using Facebook account credentials*/
            
            /*
             *code
             */
            
            expect(true).toEqual(true);
          });
          it(' 3) Login With Google ', 
           (/* REPO: https://github.com/jesseroyc/cerialize
          */) => {/* user logs in using Google account credentials*/
            
            /*
             *code
             */
            
            expect(true).toEqual(true);
          });
      });
      
    /*Page Routing*/
     
      describe('buisnessLogic/userPageRouting',
       (/* REPO: https://github.com/jesseroyc/cerialize
      */) => {/* User Login Authorization - Has Four Features */
      
          it(' 1) No Password No Routing ', 
           (/* REPO: https://github.com/jesseroyc/cerialize
          */) => {/* redirects users with no session to <LoginPage/>*/
            
            /*
             *code
             */
            
            expect(true).toEqual(true);
          });
          it(' 2) Pricing Page Accessible ', 
           (/* REPO: https://github.com/jesseroyc/cerialize
          */) => {/* non-sessions access to <PricingPage />*/
            
            /*
             *code
             */
            
            expect(true).toEqual(true);
          });
          it(' 3) Register Page Accessible ', 
           (/* REPO: https://github.com/jesseroyc/cerialize
          */) => {/* non-sessions access to <RegisterPage />*/
            
            /*
             *code
             */
            
            expect(true).toEqual(true);
          });
          it(' 4) Landing Page Accessible', 
           (/* REPO: https://github.com/jesseroyc/cerialize
          */) => {/* non-sessions access to <LandingPage />*/
            
            /*
             *code
             */
            
            expect(true).toEqual(true);
          });
      });
  });
  
  
/*COMPONENT DATA - MODELS*/
 
  describe('login page rendering, props, classes and states', () => {
    describe('components rendered properly', () => {
      it('<LoginPage /> Snapshot', () => {
        const testRenderer = renderer.create(
          <LoginPage />,
        );
        expect(testRenderer.toJSON()).toMatchSnapshot();
      });
      it('calls componentDidMount properly', () => {
        sinon.spy(LoginPage.prototype, 'componentDidMount');
        const wrapper = mount(<LoginPage />);
        expect(LoginPage.prototype.componentDidMount.calledOnce).toEqual(true);
      });
      it('allows us to set props', () => {
        const wrapper = mount(<LoginPage cardAnimaton="cardHidden" />);
        expect(wrapper.props().cardAnimaton).toEqual('cardHidden');
        wrapper.setProps({ cardAnimaton: 'LoginPage' });
        expect(wrapper.props().cardAnimaton).toEqual('LoginPage');
      });
    });
  });
  
/*COMPONENT THEMES - VIEW*/

  describe("branding, design and static labels", () => {
    const wrapper = mount(<LoginPage />);
    
    it('ensures the facebook sign in icon and button aesthetics remain unchanged', () => {
      expect(
        wrapper.containsMatchingElement(
          <span className="MuiButton-label-223">
            <i className="fab fa-facebook-square"/>
          </span>
          )
      ).toEqual(true);
    });
    it('ensures the twitter sign in icon and button aesthetics remain unchanged', () => {
      expect(
        wrapper.containsMatchingElement(
          <span className="MuiButton-label-223">
            <i className="fab fa-twitter"/>
          </span>
          )
      ).toEqual(true);
    });
    it('ensures the google sign in icon and button aesthetics remain unchanged', () => {
      expect(
        wrapper.containsMatchingElement(
          <span className="MuiButton-label-223">
            <i className="fab fa-google-plus" />
          </span>
          )
      ).toEqual(true);
    });
    it('ensures the login title label aesthetics remain unchanged', () => {
      expect(
        wrapper.containsMatchingElement(
          <span className="MuiTypography-root-150 MuiTypography-headline-155 MuiCardHeader-title-148 LoginCard-cardTitle-109">
            Login
          </span>
          )
      ).toEqual(true);
    });
    it("ensures the 'Or Be Classical' subtitle label aesthetics remain unchanged", () => {
      expect(
        wrapper.containsMatchingElement(
            <p
              className="LoginCard-cardSubtitle-110"
            >
              Or Be Classical
            </p>
          )
      ).toEqual(true);
    });
    it("ensures the 'Email...' input field label aesthetics remain unchanged", () => {
      expect(
        wrapper.containsMatchingElement(
          <label
            className="MuiFormLabel-root-274 MuiInputLabel-root-268 MuiInputLabel-formControl-269 MuiInputLabel-animated-272 CustomInput-labelRoot-252 CustomInput-labelWithAdornment-262"
            htmlFor="email"
          >
            Email...
          </label>
          )
      ).toEqual(true);
    });
    it("ensures the 'Password' input field label aesthetics remain unchanged", () => {
      expect(
        wrapper.containsMatchingElement(
          <label
            className="MuiFormLabel-root-274 MuiInputLabel-root-268 MuiInputLabel-formControl-269 MuiInputLabel-animated-272 CustomInput-labelRoot-252 CustomInput-labelWithAdornment-262"
            htmlFor="password"
          >
            Password
          </label>
          )
      ).toEqual(true);  
    });
    it("ensures the email icon aesthetics remain unchanged", () => {
      expect(
        wrapper.containsMatchingElement(
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-296 LoginPage-inputAdornmentIcon-5"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
          )
      ).toEqual(true);
    });
    it("ensures the password icon aesthetics remain unchanged", () => {
      expect(
        wrapper.containsMatchingElement(
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-296 LoginPage-inputAdornmentIcon-5"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm6-9h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM8.9 6c0-1.71 1.39-3.1 3.1-3.1s3.1 1.39 3.1 3.1v2H8.9V6zM18 20H6V10h12v10z"
            />
          </svg>
          )
      ).toEqual(true);
    });
    it("ensures the 'Lets Go' label and aesthetics remain unchanged", () => {
      expect(
        wrapper.containsMatchingElement(
          <span
            className="MuiButton-label-223"
          >
            Let's Go
          </span>
          )
      ).toEqual(true);
    });
  });
});