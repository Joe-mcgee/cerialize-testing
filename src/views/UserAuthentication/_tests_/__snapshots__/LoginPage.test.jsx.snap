// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` LoginPage Tests  login page rendering, props, classes and states components rendered properly <LoginPage /> Snapshot 1`] = `
<div
  className="LoginPage-content-1"
>
  <div
    className="LoginPage-container-2"
  >
    <div
      className="MuiGrid-typeContainer-8 MuiGrid-spacing-xs-16-30 MuiGrid-justify-xs-center-25 GridContainer-grid-7 undefined"
    >
      <div
        className="MuiGrid-typeItem-9 MuiGrid-grid-xs-12-45 MuiGrid-grid-sm-6-52 MuiGrid-grid-md-4-63 ItemGrid-grid-98 undefined"
      >
        <form>
          <div
            className="MuiPaper-root-117 MuiPaper-shadow2-121 MuiPaper-rounded-118 LoginCard-card-99 LoginPage-cardHidden-6"
          >
            <div
              className="MuiCardHeader-root-144 LoginCard-cardHeader-101 LoginCard-roseCardHeader-108"
            >
              <div
                className="MuiCardHeader-content-147"
              >
                <span
                  className="MuiTypography-root-150 MuiTypography-headline-155 MuiCardHeader-title-148 LoginCard-cardTitle-109"
                >
                  Login
                </span>
                <span
                  className="MuiTypography-root-150 MuiTypography-body1-159 MuiTypography-colorTextSecondary-172 MuiCardHeader-subheader-149 LoginCard-cardSubtitle-110"
                >
                  <button
                    className="MuiButtonBase-root-237 MuiButton-root-222 RegularButton-button-174 RegularButton-simple-190 LoginPage-customButtonClass-3 RegularButton-justIcon-195"
                    disabled={false}
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiButton-label-223"
                    >
                      <i
                        className="fab fa-facebook-square"
                      />
                    </span>
                    <span
                      className="MuiTouchRipple-root-239"
                    />
                  </button>
                  <button
                    className="MuiButtonBase-root-237 MuiButton-root-222 RegularButton-button-174 RegularButton-simple-190 LoginPage-customButtonClass-3 RegularButton-justIcon-195"
                    disabled={false}
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiButton-label-223"
                    >
                      <i
                        className="fab fa-twitter"
                      />
                    </span>
                    <span
                      className="MuiTouchRipple-root-239"
                    />
                  </button>
                  <button
                    className="MuiButtonBase-root-237 MuiButton-root-222 RegularButton-button-174 RegularButton-simple-190 LoginPage-customButtonClass-3 RegularButton-justIcon-195"
                    disabled={false}
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiButton-label-223"
                    >
                      <i
                        className="fab fa-google-plus"
                      />
                    </span>
                    <span
                      className="MuiTouchRipple-root-239"
                    />
                  </button>
                </span>
              </div>
            </div>
            <p
              className="LoginCard-cardSubtitle-110"
            >
              Or Be Classical
            </p>
            <div
              className="MuiCardContent-root-246 LoginCard-cardContent-112"
            >
              <div>
                <div
                  aria-describedby="firstname-text"
                  className="MuiFormControl-root-264 MuiFormControl-fullWidth-267 CustomInput-formControl-261 undefined "
                  onBlur={[Function]}
                  onFocus={[Function]}
                >
                  <label
                    className="MuiFormLabel-root-274 MuiInputLabel-root-268 MuiInputLabel-formControl-269 MuiInputLabel-animated-272 CustomInput-labelRoot-252 CustomInput-labelWithAdornment-262"
                    data-shrink={false}
                    htmlFor="firstname"
                  >
                    First Name..
                  </label>
                  <div
                    className="MuiInput-root-278 MuiInput-formControl-279 MuiInput-inkbar-280 CustomInput-inkbar-249 MuiInput-underline-284 CustomInput-underline-248"
                    onBlur={[Function]}
                    onFocus={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      aria-required={false}
                      autoComplete={undefined}
                      autoFocus={undefined}
                      className="MuiInput-input-287 CustomInput-input-257 "
                      defaultValue={undefined}
                      disabled={false}
                      id="firstname"
                      name={undefined}
                      onChange={[Function]}
                      onKeyDown={undefined}
                      onKeyUp={undefined}
                      placeholder={undefined}
                      readOnly={undefined}
                      required={undefined}
                      rows={undefined}
                      type="text"
                      value={undefined}
                    />
                    <div
                      className="MuiInputAdornment-root-293 MuiInputAdornment-positionEnd-295"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-296 LoginPage-inputAdornmentIcon-5"
                        color={undefined}
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M9 11.75c-.69 0-1.25.56-1.25 1.25s.56 1.25 1.25 1.25 1.25-.56 1.25-1.25-.56-1.25-1.25-1.25zm6 0c-.69 0-1.25.56-1.25 1.25s.56 1.25 1.25 1.25 1.25-.56 1.25-1.25-.56-1.25-1.25-1.25zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8 0-.29.02-.58.05-.86 2.36-1.05 4.23-2.98 5.21-5.37C11.07 8.33 14.05 10 17.42 10c.78 0 1.53-.09 2.25-.26.21.71.33 1.47.33 2.26 0 4.41-3.59 8-8 8z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  aria-describedby="email-text"
                  className="MuiFormControl-root-264 MuiFormControl-fullWidth-267 CustomInput-formControl-261 undefined "
                  onBlur={[Function]}
                  onFocus={[Function]}
                >
                  <label
                    className="MuiFormLabel-root-274 MuiInputLabel-root-268 MuiInputLabel-formControl-269 MuiInputLabel-animated-272 CustomInput-labelRoot-252 CustomInput-labelWithAdornment-262"
                    data-shrink={false}
                    htmlFor="email"
                  >
                    Email...
                  </label>
                  <div
                    className="MuiInput-root-278 MuiInput-formControl-279 MuiInput-inkbar-280 CustomInput-inkbar-249 MuiInput-underline-284 CustomInput-underline-248"
                    onBlur={[Function]}
                    onFocus={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      aria-required={false}
                      autoComplete={undefined}
                      autoFocus={undefined}
                      className="MuiInput-input-287 CustomInput-input-257 "
                      defaultValue={undefined}
                      disabled={false}
                      id="email"
                      name={undefined}
                      onChange={[Function]}
                      onKeyDown={undefined}
                      onKeyUp={undefined}
                      placeholder={undefined}
                      readOnly={undefined}
                      required={undefined}
                      rows={undefined}
                      type="text"
                      value={undefined}
                    />
                    <div
                      className="MuiInputAdornment-root-293 MuiInputAdornment-positionEnd-295"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-296 LoginPage-inputAdornmentIcon-5"
                        color={undefined}
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  aria-describedby="password-text"
                  className="MuiFormControl-root-264 MuiFormControl-fullWidth-267 CustomInput-formControl-261 undefined "
                  onBlur={[Function]}
                  onFocus={[Function]}
                >
                  <label
                    className="MuiFormLabel-root-274 MuiInputLabel-root-268 MuiInputLabel-formControl-269 MuiInputLabel-animated-272 CustomInput-labelRoot-252 CustomInput-labelWithAdornment-262"
                    data-shrink={false}
                    htmlFor="password"
                  >
                    Password
                  </label>
                  <div
                    className="MuiInput-root-278 MuiInput-formControl-279 MuiInput-inkbar-280 CustomInput-inkbar-249 MuiInput-underline-284 CustomInput-underline-248"
                    onBlur={[Function]}
                    onFocus={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      aria-required={false}
                      autoComplete={undefined}
                      autoFocus={undefined}
                      className="MuiInput-input-287 CustomInput-input-257 "
                      defaultValue={undefined}
                      disabled={false}
                      id="password"
                      name={undefined}
                      onChange={[Function]}
                      onKeyDown={undefined}
                      onKeyUp={undefined}
                      placeholder={undefined}
                      readOnly={undefined}
                      required={undefined}
                      rows={undefined}
                      type="text"
                      value={undefined}
                    />
                    <div
                      className="MuiInputAdornment-root-293 MuiInputAdornment-positionEnd-295"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-296 LoginPage-inputAdornmentIcon-5"
                        color={undefined}
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm6-9h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM8.9 6c0-1.71 1.39-3.1 3.1-3.1s3.1 1.39 3.1 3.1v2H8.9V6zM18 20H6V10h12v10z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="MuiCardActions-root-303 LoginCard-cardActions-111 LoginCard-center-116"
            >
              <button
                className="MuiButtonBase-root-237 MuiButton-root-222 RegularButton-button-174 RegularButton-roseNoBackground-188 RegularButton-wd-196 RegularButton-lg-197"
                disabled={false}
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label-223"
                >
                  Let's Go
                </span>
                <span
                  className="MuiTouchRipple-root-239"
                />
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
